File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        int n = arr.size();
        int largest = arr[0];
        int secondlargest = -1;
        
       
        
        for(int i = 1; i < n; i++){
            if(arr[i] > largest){
                secondlargest = largest;
                largest = arr[i];
            }
            else if(arr[i] > secondlargest && arr[i] < largest){
                secondlargest = arr[i];
            }
        }
        return secondlargest;
    }
};

File: Difficulty: Easy/Second Largest/second-largest.py
class Solution:
    def getSecondLargest(self, arr):
        # Code Here
        l = -float('inf')
        s = -float('inf')
        
        for i in range(len(arr)):
            if arr[i] > l:
                s = l
                l = arr[i]
            elif arr[i] < l and arr[i] > s:
                s = arr[i]
        
        return s if s != -float('inf') else -1 